beacon-chain/powchain/service.go

```diff
@@ -158,20 +206,108 @@ func (w *Web3Service) run(done <-chan struct{}) {
 				"blockHash":   w.blockHash.Hex(),
 			}).Debug("Latest web3 chain event")
 		case VRClog := <-w.logChan:
-			// public key is the second topic from validatorRegistered log.
-			pubKeyLog := VRClog.Topics[1].Hex()
-			// Support user pubKeys with or without the leading 0x.
-			if pubKeyLog == w.pubKey || pubKeyLog[2:] == w.pubKey {
-				log.WithFields(logrus.Fields{
-					"publicKey": pubKeyLog,
-				}).Info("Validator registered in VRC with public key")
-				w.validatorRegistered = true
-				w.logChan = nil
-			}
+			w.processLog(VRClog)
+
 		}
 	}
 }
 
+func (w *Web3Service) processLog(VRClog gethTypes.Log) {
+	// Process logs according to their event signature.
+	if VRClog.Topics[0] == hashutil.Hash(depositEventSignature) {
+		w.processDepositLog(VRClog)
+		return
+	}
+
+	if VRClog.Topics[0] == hashutil.Hash(chainStartEventSignature) {
+		w.processChainStartLog(VRClog)
+		return
+	}
+
+	log.Debugf("Log is not of a valid event signature %#x", VRClog.Topics[0])
+}
+
+// processDepositLog processes the log which had been received from
+// the POW chain by trying to ascertain which participant deposited
+// in the contract
+func (w *Web3Service) processDepositLog(VRClog gethTypes.Log) {
+
+	merkleRoot := VRClog.Topics[1]
+	depositData, MerkleTreeIndex, err := w.unPackLogData(VRClog.Data)
+	if err != nil {
+		log.Errorf("Could not unpack log %v", err)
+		return
+	}
+
+	if err := w.saveInTrie(depositData, merkleRoot); err != nil {
+		log.Errorf("Could not save in trie %v", err)
+		return
+	}
+
+	depositInput, err := blocks.DecodeDepositInput(depositData)
+	if err != nil {
+		log.Errorf("Could not decode deposit input  %v", err)
+		return
+	}
+
+	index := binary.BigEndian.Uint64(MerkleTreeIndex)
+
+	log.WithFields(logrus.Fields{
+		"publicKey":         depositInput.Pubkey,
+		"merkle tree index": index,
+	}).Info("Validator registered in VRC with public key and index")
+
+}
+
+func (w *Web3Service) processChainStartLog(VRClog gethTypes.Log) {
```

Comment:
```go
func (w *Web3Service) processChainStartLog(_ gethTypes.Log) {
  // TODO
}
```

Reply:
You will need to ignore the parameter name or the linter will complain it is unused. Please refer to the example I gave above.

Reply:
Nevermind, it looks like the linter doesn't care either way